migrate_pim/apply_patch/apply_patch_ee_flexibility.rst

```diff
@@ -42,6 +42,8 @@ Then clean the cache, re-install assets and warmup the cache:
 
     partners_clear_cache
     bin/console --env=prod pim:installer:assets
+    cp vendor/akeneo/pim-enterprise-dev/yarn.lock yarn.lock
```

Comment:
I don't know why we must do this, but if we don't, the webpack step will fail.

Reply:
We should not do that in my opinion  It means they can't add any library. 
The real reason is that Akeneo 4.0 does not work with an updated yarn.lock... 
Also, why the nightly build is working? AFAIK, it does not copy the yarn.lock.

Reply:
>  It means they can't add any library.

I'm not sure of that. I'm overwriting the lock file, not the json. I thought that any dependencie not present in the lock file would be downloaded.

Reply:
And it's already way better than what we tell to the customers having the issue (see related Jira ticket) :sweat_smile:

Reply:
Yes, but it's not the solution and not even a workaround in my opinion. It just hides the problem.

As soon as they want to update the yarn.lock, they will have this issue. It's not a fix. The correct fix is to understand why an update of yarn.lock fails with 4.0.x or to determine which update of dependency is triggering the error and fix the patch number in the package.json for this dependency.

And we don't copy the yarn.lock in nightly builds.

Reply:
Note: if your build is working with ".lock" but not without, you have an issue, whatever the language/project. ".lock" helps you to fix the dependency to reproduce a context. It's important to update for security purpose (or having the choice to update it).

Reply:
Note 2: 
> I'm not sure of that. I'm overwriting the lock file, not the json. I thought that any dependencie not present in the lock file would be downloaded.

It will not work. We check that the lock is consistent with the package in yarn and composer, which is a good practice. Lock file should reflect package.json (same for any dependency tool like composer, cargo, etc).